Mergin/plugin.py

```diff
@@ -278,6 +292,13 @@ def sync_project(self):
             QMessageBox.information(None, 'Project sync', 'Project is already up-to-date', QMessageBox.Close)
             return
 
+        # Check if user have writing rights to the project.
+        can_write = self._check_writing_permissions()
+        if can_write is False:
+            QMessageBox.information(None, "Project sync", "You have no writing rights to this project",
+                                    QMessageBox.Close)
+            return
+
```

Comment:
This check will also disable pull for those without write access which is not correct.

We should always let the pull to proceed and only disable push. Also, in case there are no changes to push, we do not need to show any warning to the user (and return once pull has finished).

Reply:
Should be fixed now.